TypeCobol/Compiler/Sql/CodeElements/SqlCodeElementCheckers.cs

```diff
@@ -16,10 +18,109 @@ public static class SavepointStatementChecker
+                if (!isInteger)
+                {
+                    DiagnosticUtils.AddError(alterSequenceStatement,
+                        " In minValue only digits '0' are allowed after decimal point",
+                        context);
+                }
+            }
+
+            if (alterSequenceStatement.MaxValue != null)
+            {
+                var decimalLiteral = (DecimalLiteralTokenValue)alterSequenceStatement.MaxValue.Literal.LiteralValue;
+                CheckIsInteger(alterSequenceStatement.MaxValue.Literal);
+                if (!isInteger)
+                {
+                    DiagnosticUtils.AddError(alterSequenceStatement,
+                        "In maxValue only digits '0' are allowed after decimal point",
+                        context);
+                }
+
+                if (alterSequenceStatement.MinValue != null)
+                {
+                    var minDecimalLiteral = (DecimalLiteralTokenValue)alterSequenceStatement.MinValue.Literal.LiteralValue;
+                    if (minDecimalLiteral.Number > decimalLiteral.Number)
+                    {
+                        DiagnosticUtils.AddError(alterSequenceStatement,
+                            "The maxValue must be greater than or equal to the minimum value.",
+                            context);
+                    }
+                }
+            }
+
+            if (alterSequenceStatement.RestartValue != null)
+            {
+                CheckIsInteger(alterSequenceStatement.RestartValue.Literal);
+                if (!isInteger)
+                {
+                    DiagnosticUtils.AddError(alterSequenceStatement,
+                        "In RESTART value only digits '0' are allowed after decimal point",
                         context);
                 }
             }
+
+            if (alterSequenceStatement.IncrementValue != null)
+            {
+                CheckIsInteger(alterSequenceStatement.IncrementValue.Literal);
+                if (!isInteger)
+                {
+                    DiagnosticUtils.AddError(alterSequenceStatement,
+                        "The INCREMENT value only digits '0' are allowed after decimal point",
+                        context);
+                }
+            }
+
+            if (alterSequenceStatement.CacheSize != null)
+            {
+                var tokenValue = alterSequenceStatement.CacheSize.Literal.LiteralValue;
+                var integerLiteral = (IntegerLiteralTokenValue) tokenValue;
+                if (integerLiteral.Number < 2)
+                {
+                    DiagnosticUtils.AddError(alterSequenceStatement,
+                        "Minimum value for cache size is 2",
+                        context);
+                }
+
+            }
+            void CheckIsInteger(Token optionValue)
+            {
+                var tokenValue = optionValue.LiteralValue;
+                if (tokenValue.Type != LiteralTokenValueType.Decimal) return;
```

Comment:
Ok because this method expects only DecimalLiteral or IntegerLiteral. Please add an assert before the test.

`Debug.Assert(tokenValue.Type == LiteralTokenValue.Integer || tokenValue.Type == LiteralTokenValue.Decimal);`